webapp/channels/src/components/signup/signup.tsx

```diff
@@ -564,13 +580,70 @@ const Signup = ({onCustomizeHeader}: SignupProps) => {
             }
 
             await handleSignupSuccess(user, data as UserProfile);
+            if (subscribeToSecurityNewsletter) {
+                subscribeToSecurityNewsletterFunc();
+            }
         } else {
             setIsWaiting(false);
         }
     };
 
     const handleReturnButtonOnClick = () => history.replace('/');
 
+    const getNewsletterCheck = () => {
+        if (!isCloud) {
+            if (canReachCWS) {
+                return (
+                    <CheckInput
+                        id='signup-body-card-form-check-newsletter'
+                        name='newsletter'
+                        onChange={() => setSubscribeToSecurityNewsletter(!subscribeToSecurityNewsletter)}
+                        text={
+                            formatMessage(
+                                {id: 'newsletter_optin.checkmark.text', defaultMessage: 'I would like to receive Mattermost security updates via newsletter. Data <a>Terms and Conditions</a> apply'},
+                                {
+                                    a: (chunks: React.ReactNode | React.ReactNodeArray) => (
+                                        <ExternalLink
+                                            location='signup-newsletter-checkmark'
+                                            href='https://mattermost.com/security-updates/'
+                                        >
+                                            {chunks}
+                                        </ExternalLink>
+                                    ),
+                                },
+                            )}
+                        checked={subscribeToSecurityNewsletter}
+                    />
+                );
+            }
+            return (
+                <div className='newsletter'>
+                    <span className='interested'>
+                        {formatMessage({id: 'newsletter_optin.title', defaultMessage: 'Interested in receiving Mattermost security updates via newsletter?'})}
+                    </span>
+                    <span className='link'>
+                        {formatMessage(
+                            {id: 'newsletter_optin.desc', defaultMessage: 'Sign up at <a>{link}</a>.'},
+                            {
+                                link: 'https://mattermost.com/security-updates/',
+                                a: (chunks: React.ReactNode | React.ReactNodeArray) => (
+                                    <ExternalLink
+                                        location='signup'
```

Comment:
Is it worth having distinct `location` values to the airgapped and non-airgapped scenarios? I'm ok having this branch also use an `ExternalLink` since it *is* an external link, but wanted to point out that in practice we likely won't see much telemetry from this branch because it is only reached if CWS is not reachable, which if it isn't, then likely the marketing site isn't available either (because they are probably airgapped).

Reply:
Team edition can also fail the cws availability check because the cloud interface isn't built-in.